uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request.controller.js

```diff
@@ -99,15 +99,11 @@ define([
         initRoles(),
         loadAbsencePeriods(),
         loadStatuses()
-      ]).then(function () {
-        return $q.all([
-          initAbsencePeriod(),
-          initRequest()
-        ]);
-      })
+      ])
+      .then(initRequest)
+      .then(initOpenMode)
```

Comment:
@AkA84 We have both `vm.leaveRequest` and `vm.request`, which is not ideal. If we now have an utilisation of both functions as well that would make the code even more confusing. So I think leaving it as it is now is better.

Reply:
We shouldn't sacrifice the app performance just because we didn't write good code. If those two things can be done in parallel, then we should do it. The user experience should not be penalized because we weren't able to use just one property for the request

Reply:
Done.

Reply:
@AkA84 I reviewed this again and how it was done seems to be correct to me. vm.request is always a request instance while vm.leaveRequest can be empty (in case of "create" mode). So I reverted this change back.